com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderProgramBuilder.java

```diff
@@ -487,126 +488,109 @@ static private ShaderDesc.Shader.Builder buildSpirvFromGLSL(ByteArrayInputStream
         return builder;
     }
 
-    public ShaderDesc compile(ByteArrayInputStream is, ES2ToES3Converter.ShaderType shaderType, IResource resource, String resourceOutput, String platform, boolean isDebug, boolean outputSpirv, boolean soft_fail) throws IOException, CompileExceptionError {
```

Comment:
This function was hard to expand so it's now split into more granular pieces

Reply:
Note that this will change again when the support for shader compiler hooks gets pulled in, so alot of this will be replaced / refactured anyway.